### PR TITLE
i#3544 RISC-V, part 13: Add cross config file

### PR DESCRIPTION
**Pull request by semihalf-kardach-stanislaw**
This enables cross-compiling DynamoRIO for RISC-V using Ubuntu default
toolchain prefix of riscv64-linux-gnu-. To cross-compile DynamoRIO for
RISC-V do the following:
```
  # Install a cross toolchain.
  sudo apt install g++-riscv64-linux-gnu
  # From the root of DynamoRIO
  mkdir build-riscv && cd build-riscv
  cmake -DCMAKE_TOOLCHAIN_FILE=../make/toolchain-riscv64.cmake ..
  make -j
```
Issue: #3544
Depends: #5608, #5609, #5625

Signed-off-by: Stanislaw Kardach <kda@semihalf.com>

### CODE REVIEW

File: make/toolchain-riscv64.cmake [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# search for programs in the build host directories
+SET(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+# for libraries and headers in the target directories
+SET(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+SET(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+
+# Set additional variables.
+# If we don't set some of these, CMake will end up using the host version.
+# We want the full path, however, so we can pass EXISTS and other checks in
+# the our CMake code.
```
Comment (derekbruening) [inline make/toolchain-riscv64.cmake:58]: grammar: s/the//

File: make/toolchain-riscv64.cmake [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+
+# For cross-compiling on riscv64 Linux using gcc-riscv64-linux-gnu package:
+# - install riscv64 toolchain:
+#   $ sudo apt-get install g++-riscv64-linux-gnu
```
Comment (derekbruening) [inline make/toolchain-riscv64.cmake:33]: The PR description says to use crossbuild-essential-riscv64, which is g++ plus some other tools?  I assume those other tools aren't needed.  Plus the crossbuild tool is only on Ubuntu?  So maybe update the PR description to use this g++ package to match here?

**Review by derekbruening — APPROVED** [review]

File: make/toolchain-riscv64.cmake (hunk 2 above)
Comment (semihalf-kardach-stanislaw) [reply make/toolchain-riscv64.cmake:33]: Done.

File: make/toolchain-riscv64.cmake (hunk 1 above)
Comment (semihalf-kardach-stanislaw) [reply make/toolchain-riscv64.cmake]: Done.

**Pull request merged**